### PR TITLE
Add get all adverts by category

### PR DESCRIPTION
**Pull request by Elder-HD**

### CODE REVIEW

File: src/main/java/com/codegym/jrugotom5/controller/AdvertController.java [hunk 1]
```diff
@@ -2,29 +2,32 @@
...
         return advertService.getAllAdverts();
     }
 
-    @GetMapping(path = "/advert")
+    @GetMapping(path = "/date")
     public List<AdvertDTO> getAdverts(@RequestParam LocalDate from, LocalDate to) {
         return advertService.getAdvertsByDateRange(from, to);
     }
+
+    @GetMapping("/{category}/")
```
Comment (Volodymyr95) [inline src/main/java/com/codegym/jrugotom5/controller/AdvertController.java:37]: make category as enum

Comment (Elder-HD) [reply src/main/java/com/codegym/jrugotom5/controller/AdvertController.java:37]: Done

File: src/main/java/com/codegym/jrugotom5/controller/AdvertController.java [hunk 2]
```diff
@@ -27,4 +26,14 @@ public List<AdvertBasicInfoDTO> getAllAdverts() {
     public List<AdvertFullInfoDTO> getAdverts(@RequestParam LocalDate from, LocalDate to) {
         return advertService.getAdvertsByDateRange(from, to);
     }
+
+    @GetMapping("/{category}/")
+    public List<AdvertBasicInfoDTO> getByCategory(@PathVariable String category) {
+        try {
+            Category enumCategory = Category.valueOf(category.toUpperCase());
```
Comment (Volodymyr95) [inline src/main/java/com/codegym/jrugotom5/controller/AdvertController.java]: validation should be on service layer, if it's not a annotation validation

File: src/main/java/com/codegym/jrugotom5/exception/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
+@Slf4j
 public class GlobalExceptionHandler {
     @ExceptionHandler(InvalidDateRangeException.class)
     public ResponseEntity<String> handleInvalidDateRange(InvalidDateRangeException ex) {
         return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);
     }
+
+    @ExceptionHandler(InvalidCategoryException.class)
+    public ResponseEntity<String> handleInvalidCategoryException(InvalidCategoryException e) {
+        log.error("Occurred Exception: {} ", e.getMessage());
```
Comment (Volodymyr95) [inline src/main/java/com/codegym/jrugotom5/exception/GlobalExceptionHandler.java]: is message enought?

File: src/main/java/com/codegym/jrugotom5/controller/AdvertController.java (hunk 2 above)
Comment (Elder-HD) [reply src/main/java/com/codegym/jrugotom5/controller/AdvertController.java]: Fixed: Move Validation to service layer

File: src/main/java/com/codegym/jrugotom5/exception/GlobalExceptionHandler.java (hunk 3 above)
Comment (Elder-HD) [reply src/main/java/com/codegym/jrugotom5/exception/GlobalExceptionHandler.java]: Added error objetct to log message

**Comment by Elder-HD** [comment]
Please check but don't merge till i add flyway migration for Category field

**Review by Volodymyr95 — APPROVED** [review]

**Pull request merged**